esmf_regrid/experimental/io.py

```diff
@@ -52,11 +52,18 @@ def save_regridder(rg, filename):
     regridder_type = rg.__class__.__name__
     if regridder_type == "GridToMeshESMFRegridder":
```

Comment:
Given that as far as I can tell we do equivalent steps whichever way around the source and target are, we could build the mesh cube and the grid cube and then assign their names based on what type of regridder we're in? It would reduce code duplication, but might tell it slightly trickier to tell what's going on, so I don't mind strongly either way.